dotnet/src/Agents/Core/Chat/TerminationStrategy.cs

```diff
@@ -3,14 +3,17 @@
 using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
 
 namespace Microsoft.SemanticKernel.Agents.Chat;
 
 /// <summary>
 /// Base strategy class for defining termination criteria for a <see cref="AgentGroupChat"/>.
 /// </summary>
-public abstract class TerminationStrategy
+public abstract class TerminationStrategy(ILogger logger)
```

Comment:
Can we please not modify the constructor contract?  (please use an init property)

Constructor parameters bubble through every subclass.

Reply:
I was struggling with this. I don't know if we should have the logger as a public property. Maybe we can make the logger argument optional and default to null?